xbmc/windowing/wayland/WinSystemWayland.cpp

```diff
@@ -833,15 +833,15 @@ void CWinSystemWayland::SetResolutionInternal(CSizeInt size, std::int32_t scale,
         XBMC_Event msg{XBMC_MODECHANGE};
         msg.mode.res = RES_WINDOW;
         SetWindowResolution(sizes.bufferSize.Width(), sizes.bufferSize.Height());
-        CWinEvents::MessagePush(&msg);
+        dynamic_cast<CWinEventsWayland*>(m_winEvents)->MessagePush(&msg);
```

Comment:
dynamic_cast can return `nullptr`, i suggest changing to static_cast

Reply:
you can't use static_cast for casting interfaces. have you ever tried a static cast on an object that was derived from multiple base classes?

Reply:
both static and dynamic cast hit the vtable, regardless of the cast type. Casting to wayland events outside of wayland is a programmer error, so there's no need for RTTI to cause a segfault in this case

Reply:
I know that static cast fails in some cases because I already fell into that trap. 

class A : public B, public C
when casting from i.e. B to C you either have to go through A, when using static_cast, or you use dynamic_cast.

anyway, a better solution for wayland would be to maintain a member with correct type. I leave this to @pkerling

Reply:
Do we even still need a common `CWinEvents` at all? Shouldn't it be an implementation detail of the windowing implementation in question how it wants to treat the event stuff? Then the cast also would not be necessary.

Reply:
In this iteration we still poll MessagePump through the interface. Later when windowing will own the render thread this can be dropped and windowing can send the events to application directly.

Reply:
Can we leave it like this for this iteration then and add a FIXME? I don't think it makes much sense to add a workaround now if it will go away in any case.

If the NULL pointer thing is still a concern, you can do `dynamic_cast<CWinEventsWayland&>(*m_winEvents).MessagePush` instead.

Reply:
ok